### PR TITLE
Wrong use of return type "NoReturn"

### PR DESCRIPTION
**Pull request by thunfisch987**
### Description of the problem

In a lot of the type hinted functions, the return type hint "NoReturn" was wrongly used.
It's supposed to be used with functions that (under all conditions) _do not_ return.
Example for a NoReturn function:
```python
def stop() -> NoReturn:
    raise RuntimeError()
```

### Reproducing the problem

An Example of where it was used wrongly:
```python
def set_active_underline_width(self, width: Union[float, int]) -> NoReturn:
    """Animates the width of the active underline line."""

    Animation( _underline_width=width, duration=(0.2 if self.line_anim else 0), t="out_quad" ).start(self)
```
[Animation(...).start() always returns None] 

### Description of Changes

Every occasion of NoReturn was checked and changed to None, as all of them were wrong.

### CODE REVIEW

**Comment by thunfisch987** [comment]
there's also a wrong type hint in datatables.py which said `NoReturn` which i changed to `None`, but should be `bool`

File: kivymd/toast/kivytoast/kivytoast.py [hunk 1]
```diff
@@ -38,7 +38,7 @@ def build(self):
     Test().run()
 """
 
-from typing import List, NoReturn
+from typing import List, None
```
Comment (HeaTTheatR) [inline kivymd/toast/kivytoast/kivytoast.py]:
```python
from typing import List, None
```

```
  File "<input>", line 1
    from typing import List, None
                             ^
SyntaxError: trailing comma not allowed without surrounding parentheses
```

File: kivymd/tools/patterns/create_project.py [hunk 2]
```diff
@@ -228,37 +228,37 @@ def __init__(self, model):
...
+    def on_tap_button_login(self) -> None:
         """Called when the `LOGIN` button is pressed."""
 
-    def set_user_data(self, key, value) -> NoReturn:
+    def set_user_data(self, key, value) -> None:
         """Called every time the user enters text into the text fields."""
 '''
 
-_firebase_view_import = """from typing import Union, NoReturn
+_firebase_view_import = """from typing import Union, None
```
Comment (HeaTTheatR) [inline kivymd/tools/patterns/create_project.py]:
```python
from typing import List, None
```

```
  File "<input>", line 1
    from typing import List, None
                             ^
SyntaxError: trailing comma not allowed without surrounding parentheses
```

File: kivymd/uix/behaviors/backgroundcolor_behavior.py [hunk 3]
```diff
@@ -7,7 +7,7 @@
 
 __all__ = ("BackgroundColorBehavior", "SpecificBackgroundColorBehavior")
 
-from typing import List, NoReturn
+from typing import List, None
```
Comment (HeaTTheatR) [inline kivymd/uix/behaviors/backgroundcolor_behavior.py]:
```python
from typing import List, None
```

```
  File "<input>", line 1
    from typing import List, None
                             ^
SyntaxError: trailing comma not allowed without surrounding parentheses
```

File: kivymd/uix/filemanager/filemanager.py [hunk 4]
```diff
@@ -133,7 +133,7 @@ def file_manager_open(self):
 import locale
 import os
 import re
-from typing import List, NoReturn, Tuple, Union
+from typing import List, None, Tuple, Union
```
Comment (HeaTTheatR) [inline kivymd/uix/filemanager/filemanager.py]:
```python
from typing import List, None
```

```
  File "<input>", line 1
    from typing import List, None
                             ^
SyntaxError: trailing comma not allowed without surrounding parentheses
```

File: kivymd/uix/pickers/colorpicker/colorpicker.py [hunk 5]
```diff
@@ -14,7 +14,7 @@
 
 .. code-block:: python
 
-    from typing import Union, NoReturn
+    from typing import Union, None
```
Comment (HeaTTheatR) [inline kivymd/uix/pickers/colorpicker/colorpicker.py]:
```python
from typing import List, None
```

```
  File "<input>", line 1
    from typing import List, None
                             ^
SyntaxError: trailing comma not allowed without surrounding parentheses
```

File: kivymd/uix/pickers/colorpicker/colorpicker.py [hunk 6]
```diff
@@ -76,7 +76,7 @@ def on_select_color(self, instance_gradient_tab, color: list) -> NoReturn:
 import os
 import struct
 from io import BytesIO
-from typing import List, NoReturn, Union
+from typing import List, None, Union
```
Comment (HeaTTheatR) [inline kivymd/uix/pickers/colorpicker/colorpicker.py]:
```python
from typing import List, None
```

```
  File "<input>", line 1
    from typing import List, None
                             ^
SyntaxError: trailing comma not allowed without surrounding parentheses
```

**Comment by thunfisch987** [comment]
Oh....
Looks like there were some imports i overlooked😬
(Since this is my first commit, how do i fix it? Do i open a new one with the fixed code?)

**Comment by HeaTTheatR** [comment]
@thunfisch987 Make another push.

File: kivymd/uix/templates/scalewidget/scalewidget.py [hunk 7]
```diff
@@ -15,7 +15,7 @@
 
 .. code-block:: python
 
-    
+
```
Comment (HeaTTheatR) [inline kivymd/uix/templates/scalewidget/scalewidget.py]: There must be only one newline

File: kivymd/uix/templates/rotatewidget/rotatewidget.py [hunk 8]
```diff
@@ -15,7 +15,7 @@
 
 .. code-block:: python
 
-    
+
```
Comment (HeaTTheatR) [inline kivymd/uix/templates/rotatewidget/rotatewidget.py]: There must be only one newline

File: kivymd/uix/templates/scalewidget/scalewidget.py (hunk 7 above)
Comment (thunfisch987) [reply kivymd/uix/templates/scalewidget/scalewidget.py]: sorry, forgot that pre-commit does not check the docstrings

File: kivymd/tools/patterns/create_project.py [hunk 9]
```diff
@@ -179,7 +179,7 @@ class {name_screen}Model(BaseScreenModel):
     :class:`~View.{module_name}.{name_screen}.{name_screen}View` class.
     """'''
 
-_firebase_controller = '''from typing import NoReturn
+_firebase_controller = '''
 {import_module}
```
Comment (thunfisch987) [inline kivymd/tools/patterns/create_project.py:183]: is this still needed?

File: kivymd/tools/patterns/create_project.py [hunk 10]
```diff
@@ -197,22 +197,22 @@ def __init__(self, model):
...
         self.model.chek_data()
 
-    def reset_data_validation_status(self, *args) -> NoReturn:
+    def reset_data_validation_status(self, *args) -> None:
         self.model.reset_data_validation_status()
 '''
 
-_without_firebase_controller = '''from typing import NoReturn
+_without_firebase_controller = '''
 {import_module}
```
Comment (thunfisch987) [inline kivymd/tools/patterns/create_project.py:216]: and this one

File: kivymd/tools/patterns/create_project.py (hunk 9 above)
Comment (thunfisch987) [reply kivymd/tools/patterns/create_project.py:183]: i mean the {import_module}

Comment (thunfisch987) [reply kivymd/tools/patterns/create_project.py:183]: because i don't know anything about firebase

**Pull request merged**